File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count = Counter()
        max_w_freq=0
        max_freq=0

        left =0
        for right in range(len(s)):
            count[s[right]]+=1
            max_freq = max(max_freq,count[s[right]])
            if((right-left+1)-max_freq)>k:
                count[s[left]]-=1
                left +=1
            max_w_freq =max(max_w_freq,right-left+1)
        return max_w_freq

      

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.java
class Solution {
    public int characterReplacement(String s, int k) {
        int[] counter = new int[26];
        int i=0;
        int max_freq=0;
        int max_count=0;
        for(int j=0;j<s.length();j++)
        {  char ch = s.charAt(j);
            counter[ch-'A']++;
            max_count = Math.max(max_count,counter[ch-'A']);
            while(((j-i+1)-max_count)>k)
            {
                counter[s.charAt(i)-'A']--;
                i++;
            }
            max_freq=Math.max(max_freq,(j-i+1));
        }
        return max_freq;
    }
}